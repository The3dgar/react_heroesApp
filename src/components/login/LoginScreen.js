import React from 'react'

export const LoginScreen = ({history}) => {
  const handleClick = () => {
    history.replace("/")
  }

  return (
    <div className="container mt-5">
      <h2>LoginScreen</h2>
      <hr/>
      <button 
        className="btn btn-primary"
        onClick={handleClick}
      >
        Login
      </button>

    </div>
  )
}
