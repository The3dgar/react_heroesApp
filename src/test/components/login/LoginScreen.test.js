import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";

describe("Test in <LoginScreen/>", () => {
  const context = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  const history = {
    length: 10,
    goBack: jest.fn(),
    push: jest.fn(),
    replace: jest.fn()
  };

  const wrapper = mount(
    <AuthContext.Provider value={context}>
      <LoginScreen history={history} />
    </AuthContext.Provider>
  );

  test("should displayed correctly", () => {
    expect(wrapper).toMatchSnapshot()

  });


  test('should do dispatch and navigation', () => {
    const handleClick = wrapper.find("button").prop("onClick")


    handleClick()

    expect(context.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload:{
        name: "Edgar"
      }
    })
    // expect(history.replace).toHaveBeenCalled()
    expect(history.replace).toHaveBeenCalledWith("/")
    localStorage.setItem("lastPath", "/dc")
    handleClick()
    expect(history.replace).toHaveBeenCalledWith("/dc")
  })
  
});
