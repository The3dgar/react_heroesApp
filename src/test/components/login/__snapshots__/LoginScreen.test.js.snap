// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen/> should displayed correctly 1`] = `
<div
  className="container mt-5"
>
  <h2>
    LoginScreen
  </h2>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
